--- calfw-org.el.bak	2022-02-11 15:32:07.428078355 +0100
+++ calfw-org.el	2022-02-11 15:36:14.241391455 +0100
@@ -6,6 +6,7 @@
 ;; Keywords: calendar, org
 ;; Package-Version: 20160303.258
 ;; Package-Commit: 03abce97620a4a7f7ec5f911e669da9031ab9088
+;; with patch from https://github.com/kiwanami/emacs-calfw/pull/134/files
 
 ;; This program is free software; you can redistribute it and/or modify
 ;; it under the terms of the GNU General Public License as published by
@@ -240,14 +241,13 @@
 				(match-string 1 extra)))
 		      (total-days (string-to-number
 				   (match-string 2 extra)))
-		      (start-date (time-subtract
-				   (org-read-date nil t date-string)
-				   (seconds-to-time (* 3600 24 (- cur-day 1)))))
+		      (start-date (org-read-date nil t date-string))
 		      (end-date (time-add
-				 (org-read-date nil t date-string)
-				 (seconds-to-time (* 3600 24 (- total-days cur-day))))))
-		 (list (calendar-gregorian-from-absolute (time-to-days start-date))
-		       (calendar-gregorian-from-absolute (time-to-days end-date)) text))
+				 start-date
+				 (seconds-to-time (* 3600 24 (- total-days 1))))))
+		       (unless (= cur-day total-days)
+             (list (calendar-gregorian-from-absolute (time-to-days start-date))
+		                  (calendar-gregorian-from-absolute (time-to-days end-date)) text)))
 	     )))))
 
 (defun cfw:org-schedule-period-to-calendar (begin end)
